Dart/src/com/jetbrains/lang/dart/resolve/DartResolver.java

```diff
@@ -189,6 +185,33 @@ else if (!lValue && component.isGetter()) {
     });
   }
 
+  /**
+   * Find the region with the given offset in the given list of sorted regions.
+   * Returns the found region or null.
+   */
+  @Nullable
+  private static PluginNavigationRegion findRegionAtOffset(List<PluginNavigationRegion> regions, int offset) {
```

Comment:
Before this change DartResolver.resolve() was 11% of total time.

Now DartResolver.resolve() is about 6% of total time when I type several characters with small delays and then backspace them one by one with small delays. And findRegion() itself is invisible now.

  Here is break down of the current DartResolver.resolve() state.

<img width="1132" alt="screen shot 2015-07-08 at 10 08 36 am" src="https://cloud.githubusercontent.com/assets/384794/8576792/5a1b16f0-2559-11e5-848a-d66199020d90.png">